src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml

```diff
@@ -409,3 +409,11 @@ services:
       - '@prestashop.core.query.doctrine_search_criteria_applicator'
       - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
     public: true
+
+  prestashop.core.grid.query_builder.state:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Query\StateQueryBuilder'
+    parent: 'prestashop.core.grid.abstract_query_builder'
+    public: true
+    arguments:
+      - '@prestashop.core.query.doctrine_search_criteria_applicator'
+      - "@=service('prestashop.adapter.legacy.context').getContext().employee.id_lang"
```

Comment:
Curious question but why not use `@=service("prestashop.adapter.legacy.context").getLanguage().id` here?

Relying on employee field can be a bit unstable in test environment

Reply:
With the `employee.id_lang"`, you will have the list of countries in the language of the employee.